apollo-integration/src/test/java/com/apollographql/apollo/ApolloCancelCallTest.java

```diff
@@ -46,30 +49,27 @@ public void cancelCallBeforeEnqueueCanceledException() throws Exception {
 
     Rx2Apollo.from(call)
         .test()
-        .awaitDone(TIME_OUT_SECONDS, TimeUnit.SECONDS)
         .assertError(ApolloCanceledException.class);
   }
 
   @Test
   public void cancelCallAfterEnqueueNoCallback() throws Exception {
-    server.enqueue(mockResponse("EpisodeHeroNameResponse.json")
-        .setBodyDelay(TIME_OUT_SECONDS / 2, TimeUnit.SECONDS));
-
+    OkHttpClient okHttpClient = new OkHttpClient.Builder()
+        .dispatcher(new Dispatcher(Utils.immediateExecutorService()))
+        .build();
+    apolloClient = ApolloClient.builder()
+        .serverUrl(server.url("/"))
+        .okHttpClient(okHttpClient)
+        .httpCache(new ApolloHttpCache(cacheStore, null))
+        .build();
```

Comment:
The same issue as above: `.dispatcher(Utils.immediateExecutor())`

Reply:
Actually it wasn't a coincidence that I left it here without `immediateExecutor`. 
We cannot just add here `immediateExecutor` because when doing all sequentially the values will be emmited before cancelling the call, so the assertions will fail. What we need here is an executor which can be controlled, and can fire events like `TestScheduler`.  I tried to write something like that but without luck. So for now at least I think that it has to stay as it is.